### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by xhsien**
Add tests in Model component

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PhoneTest.java [hunk 1]
```diff
@@ -18,6 +18,7 @@ public void isValidPhone() {
         assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
 
         // valid phone numbers
+        assertTrue(Phone.isValidPhone(null));
```
Comment (chilipadiboy) [inline src/test/java/seedu/address/model/person/PhoneTest.java:21]: i think these tests are good.  However maybe we can take it one step further by allowing area codes? E.g. +6597725960 etc.

**Comment by Jemereny** [comment]
Tests are alright.

File: src/test/java/seedu/address/model/person/WebsiteTest.java [hunk 2]
```diff
@@ -9,16 +9,16 @@
...
     @Test
     public void isValidWebsite() {
-        // invalid Website
+        // invalid Websites
         assertFalse(Website.isValidWebsite("")); // empty string
         assertFalse(Website.isValidWebsite(" ")); // spaces only
         assertFalse(Website.isValidWebsite("www.yahoo.com")); // no http protocol
         assertFalse(Website.isValidWebsite("http://YAHOO.com")); // non-numeric
         assertFalse(Website.isValidWebsite("http://www.YAHOO.com")); // lower-case
         assertFalse(Website.isValidWebsite("9312 1534")); // no digits
```
Comment (chilipadiboy) [inline src/test/java/seedu/address/model/person/WebsiteTest.java:18]: the comment should be either "no spaces" or "wrong format".

 assertFalse(Website.isValidWebsite("www.yahoo.com")); // no http protocol
          assertFalse(Website.isValidWebsite("http://YAHOO.com"));

the 2 websites are still valid. This test is not correct unless it strictly to enforce making the user input the "http" and "www"

File: src/test/java/systemtests/AddCommandSystemTest.java [hunk 3]
```diff
@@ -229,8 +261,13 @@ assert getModel().getFilteredPersonList().size()
 
         /* Case: invalid birthday -> rejected */
         command = AddCommand.COMMAND_WORD + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY
-                + INVALID_BIRTHDAY_DESC + WEBSITE_DESC_AMY + INVALID_TAG_DESC;
```
Comment (chilipadiboy) [inline src/test/java/systemtests/AddCommandSystemTest.java:232]: agreed its a better test

File: src/test/java/systemtests/AddCommandSystemTest.java [hunk 4]
```diff
@@ -190,14 +215,21 @@ assert getModel().getFilteredPersonList().size()
...
         assertCommandSuccess(command, toAdd);
 
+        /* Case: missing birthday -> added */
+        toAdd = new PersonBuilder().withName("Adeep").withPhone(VALID_PHONE_AMY).withEmail(VALID_EMAIL_AMY)
+            .withAddress(VALID_ADDRESS_AMY).withBirthday(EMPTY_BIRTHDAY).withWebsite(VALID_WEBSITE_AMY)
+            .withTags(VALID_TAG_FRIEND).build();
+        command = AddCommand.COMMAND_WORD + " n/Adeep" + PHONE_DESC_AMY + EMAIL_DESC_AMY
+            + ADDRESS_DESC_AMY + WEBSITE_DESC_AMY + TAG_DESC_FRIEND;
+        assertCommandSuccess(command, toAdd);
+
```
Comment (chilipadiboy) [inline src/test/java/systemtests/AddCommandSystemTest.java:232]: good test to make sure bday is optional

File: src/test/java/seedu/address/model/person/WebsiteTest.java (hunk 2 above)
Comment (chilipadiboy) [reply src/test/java/seedu/address/model/person/WebsiteTest.java:18]: however i understand that the issue is that the java browser cannot handle urls without protocols yet.

**Pull request merged**